src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        joinedString += separator;
+      } else {
+        joinedString += this[i];
```

Comment:
Ти можеш прибрати одну умову, щоб не повторюватися
```suggestion
      if (this[i] !== undefined || this[i] !== null) {
        joinedString += this[i];
```